### DISCUSSION TITLE
OpenSSL 3 Connection Issues

### DISCUSSION THREAD

**Discussion by codiverse** [Keyfactor/ejbca-ce#309] [Q&A] [answered]
We're using EJBCA 6.15.2 and have recently started seeing connection issues as clients upgrade to OpenSSL 3 (eg. SVN via MacOS Homebrew).

The openssl error message is:
> 00DE1CE201000000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:341:Global default library context, Algorithm (RC2-40-CBC : 0), Properties () 

We think it's due to that version generating certificates using pbeWithSHA1And40BitRC2-CBC, and we can work around with openssl 3 conversion of the original EJBCA issued certificate, eg.
```
openssl pkcs12 -legacy -in ./current.p12 -out ./new.pem
openssl pkcs12 -export -in ./new.pem -out ./new.p12 
```

We've also noted that there is an option in newer versions of EJBCA specifically relating to this via the `keystore.use_legacy_pkcs12` property, with the description:

> Use Legacy PKCS#12 Keystore
> Set to true to keep the implementation used by EJBCA <7.5 and encrypt certificates in PKCS#12 keystores using
> pbeWithSHA1And40BitRC2-CBC instead of pbeWithSHA1And3-KeyTripleDES-CBC. Default is false.

And we are in the process of upgrading, but aren't quite ready yet, so the question is - **can we get an OpenSSL 3 compatible encryption with EJBCA 6.15.2?**

**Comment by mike-agrenius-kushner** [comment, marked as answer]
Hi, 

Honestly 6.15 is so ancient that we don't even test on it anymore – we have EJBCA 8 CE going public in a few days, so 7.12 is the last version we currently even perform any support on.

So the simple answer is that I don't know, I'd very strongly recommend pushing on with the upgrade instead – it also contains fixes for a whole bunch of CVE's that haven't been backported to version 6. 

Cheers,
Mike

**Reply by codiverse** [reply]
Ok, thanks for the clarification, and noting the timing of the version 8 release.
